File: herencia/coworking-csharp/Course.OOP.Inheritance.Business/Auditorium.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Course.OOP.Inheritance.Business
{
    public class Auditorium: MeetingRoom
    {

        public Auditorium(String name, int chairs) : base(name, chairs) { }

        /*
        * Como el método en la clase padre no es abstracto, es opcional sobreescribirlo aquí.
        * */
        public override decimal CalculatePricePerDay()
        {
            Console.WriteLine("Ejecutando método en Auditorium");
            //Agregar un valor adicional.
            decimal basePrice = base.CalculatePricePerDay();
            decimal result = basePrice + 100;

            return result;
        }

    }
}

File: herencia/coworking-java/src/main/java/course/oop/inheritance/Auditorium.java
package course.oop.inheritance;

import java.math.BigDecimal;

public class Auditorium extends MeetingRoom {

    public Auditorium(String name, int chairs) {
        super(name, chairs);
    }

    /*
    * Como el método en la clase padre no es abstracto, es opcional sobreescribirlo aquí.
    * */
    public BigDecimal calculatePricePerDay() {

        //Agregar un valor adicional.
        BigDecimal basePrice = super.calculatePricePerDay();
        BigDecimal result = basePrice.add(new BigDecimal(100));

        return result;
    }
}
